frontend/src/concepts/modelCatalog/useModelCatalogSources.ts

```diff
@@ -20,18 +20,34 @@ export const useModelCatalogSources = (): FetchState<State> => {
       throw new NotReadyError('Model catalog feature is not enabled');
     }
 
-    const configMap = await getConfigMap(dashboardNamespace, MODEL_CATALOG_SOURCE_CONFIGMAP);
-
-    if (!configMap.data?.modelCatalogSource) {
-      return [];
-    }
-
     try {
-      const source: ModelCatalogSource = JSON.parse(configMap.data.modelCatalogSource);
-      return [source];
-    } catch (e) {
-      // Swallow JSON parse errors and return empty array for temporary summit implementation
-      return [];
+      const configMap = await getConfigMap(dashboardNamespace, MODEL_CATALOG_SOURCE_CONFIGMAP);
+      /* eslint-enable @typescript-eslint/no-unsafe-member-access */
+
+      if (!configMap.data || !configMap.data.modelCatalogSource) {
+        return [];
+      }
+
+      try {
+        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+        const source = JSON.parse(configMap.data.modelCatalogSource) as ModelCatalogSource;
+        return [source];
+      } catch (e) {
+        // Swallow JSON parse errors and return empty array
+        return [];
+      }
+    } catch (e: unknown) {
+      // Check if error is a 404
+      if (
+        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+        'statusObject' in (e as object) &&
+        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+        isK8sStatus((e as { statusObject: unknown }).statusObject)
+      ) {
+        return [];
+      }
```

Comment:
You don't need the linter overrides here either, as long as you have enough guards to narrow the type down tightly enough you won't need the `as` keyword, which we generally want to avoid like the plague as that really has the effect of saying to the type system: "I've got this, kick back and chill for a bit", so we're losing compile time type safety. 

If you actually narrow the type through conditional statements rather than using `as`, you're proving to the compiler the code is type safe, so any unexpected runtime values will be handled.

Try something like this, it's not pretty, but it's type safe and avoids the linter overrides:

```suggestion
      // Check if error is a 404
      if (
        typeof e === 'object' &&
        e != null &&
        'statusObject' in e &&
        isK8sStatus(e.statusObject)
      ) {
        return [];
      }
```